src/lib.rs

```diff
@@ -226,26 +223,33 @@ pub fn run(
             };
             if let Err(_err) = result {
                 match inexplicit_panic(&mut vm, storage) {
-                    Ok(false) => {
-                        vm.current_frame_mut()?.pc += 1;
-                        continue;
-                    }
+                    Ok(false) => continue,
                     _ => return Ok((ExecutionOutput::Panic, vm)),
                 }
             }
+            set_pc(&mut vm, &opcode)?;
+        } else {
+            vm.current_frame_mut()?.pc += 1;
         }
-        vm.current_frame_mut()?.pc = opcode_pc_set(&opcode, vm.current_frame()?.pc);
     }
     let result = retrieve_result(&mut vm)?;
     Ok((ExecutionOutput::Ok(result), vm))
 }
 
-// Set the next PC according to the next opcode
-fn opcode_pc_set(opcode: &Opcode, current_pc: u64) -> u64 {
-    match opcode.variant {
+// Sets the next PC according to the next opcode
+fn set_pc(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmError> {
+    let current_pc = vm.current_frame()?.pc;
+
+    vm.current_frame_mut()?.pc = match opcode.variant {
         Variant::FarCall(_) => 0,
+        Variant::Ret(RetOpcode::Revert) => current_pc,
+        Variant::Ret(RetOpcode::Panic) => current_pc,
+        Variant::NearCall(_) => current_pc,
+        Variant::Jump(_) => current_pc,
```

Comment:
[Multiple match patterns may be joined with the | operator](https://doc.rust-lang.org/reference/expressions/match-expr.html#:~:text=Multiple%20match%20patterns%20may%20be%20joined%20with%20the%20%7C%20operator.%20Each%20pattern%20will%20be%20tested%20in%20left%2Dto%2Dright%20sequence%20until%20a%20successful%20match%20is%20found.)

```rs
        Variant::Ret(RetOpcode::Revert)
        | Variant::Ret(RetOpcode::Panic)
        | Variant::NearCall(_)
        | Variant::Jump(_) => current_pc,
```

Reply:
Should I change it? This changes in #147 though

Reply:
It's just another way to do it, In my opinion you don't need to change it unless you want to.